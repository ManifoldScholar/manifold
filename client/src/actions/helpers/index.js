export { default as createApiAction } from "./createApiAction";
export { default as commonActions } from "./commonActions";
