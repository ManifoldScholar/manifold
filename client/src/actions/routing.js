import { createAction } from "redux-actions";

export const update = createAction("ROUTE_UPDATE", state => state);
