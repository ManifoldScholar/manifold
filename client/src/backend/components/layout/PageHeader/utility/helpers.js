import lh from "helpers/linkHandler";

export const getTextLinks = ({ texts, pathname }) =>
  texts?.length
    ? texts?.map(t => ({
        label: t.label,
        route: "backendTextAnalytics",
        id: t.id,
        active: pathname?.includes(t.id)
      }))
    : null;

export const getIssueLinks = ({ issues, pathname }) =>
  issues?.length
    ? issues?.map(i => ({
        label: i.label,
        route: "backendProjectAnalytics",
        id: i.id,
        active: pathname?.includes(i.id)
      }))
    : null;

export const getLinkOrButtonProps = action => {
  if (action.href) {
    return action.download
      ? { as: "a", href: action.href, download: true }
      : {
          as: "a",
          href: action.href,
          target: "_blank",
          re: "noopener noreferrer"
        };
  }

  if (action.route)
    return {
      to: lh.link(action.route, action.slug, action.resourceSlug)
    };

  return { as: "button", onClick: action.onClick };
};
