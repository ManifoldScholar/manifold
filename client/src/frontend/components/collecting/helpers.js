import flatMapDepth from "lodash/flatMapDepth";
import identity from "lodash/identity";
import has from "lodash/has";
import get from "lodash/get";
import isEmpty from "lodash/isEmpty";

export function getEntityCollection(entity, relationship = "collection") {
  return get(entity, `relationships.${relationship}`);
}

function isCollection(possibleCollection) {
  return has(possibleCollection, "attributes.categoryMappings");
}

function hasCollection(possibleEntity, relationship = "collection") {
  return has(possibleEntity, `relationships.${relationship}`);
}

export function collectedIdsForCollection(collection) {
  if (!collection) return [];
  const mappings = collection.attributes.categoryMappings;
  return flatMapDepth(
    Object.values(mappings).map(mapping => Object.values(mapping)),
    identity,
    2
  );
}

export function inCollection(collectionOrEntity, collectable) {
  const collection = isCollection(collectionOrEntity)
    ? collectionOrEntity
    : hasCollection(collectionOrEntity) &&
      getEntityCollection(collectionOrEntity);
  if (!collection) return false;

  const collectedIds = collectedIdsForCollection(collection);
  return collectedIds.includes(collectable.id);
}

export function inCollections(collectable, ...collectionsOrEntities) {
  return collectionsOrEntities.some(collectionOrEntity =>
    inCollection(collectionOrEntity, collectable)
  );
}

export function getResponse(id, responses) {
  return responses.find(res => res.id === id);
}

export function idInResponses(id, responses) {
  if (responses?.length < 1) return false;
  return responses.map(res => res.id).includes(id);
}

export function hasItemsInCollection(entity) {
  const collection = getEntityCollection(entity);
  if (!collection?.attributes) return false;
  return !isEmpty(collection.attributes.categoryMappings);
}
