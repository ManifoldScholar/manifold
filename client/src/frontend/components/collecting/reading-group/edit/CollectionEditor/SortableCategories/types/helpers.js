import PropTypes from "prop-types";

export const collectedShape = {
  categoryId: PropTypes.string.isRequired,
  collectedIds: PropTypes.array.isRequired,
  responses: PropTypes.array.isRequired,
  onRemove: PropTypes.func.isRequired,
  showDropzone: PropTypes.bool
};
