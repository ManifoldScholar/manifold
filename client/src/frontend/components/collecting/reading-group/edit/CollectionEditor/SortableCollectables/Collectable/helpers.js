export function getCollectableIcon(type) {
  switch (type) {
    case "projects":
      return "projects64";
    case "texts":
      return "textsLoosePages64";
    case "textSections":
      return "toc64";
    case "resourceCollections":
      return "resourceCollection64";
    case "resources":
      return "resources64";
    case "journalIssues":
      return "journals64";
    default:
      return "resources64";
  }
}
