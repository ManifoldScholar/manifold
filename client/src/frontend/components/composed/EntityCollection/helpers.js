import isEmpty from "lodash/isEmpty";

export function getHeroImage(collection) {
  const { heroLayout, heroStyles } = collection.attributes;
  if (isEmpty(heroStyles)) return null;
  switch (heroLayout) {
    case "full_bleed":
    case "wide_inset":
      return heroStyles.largeLandscape;
    case "square_inset":
      return heroStyles.mediumSquare;
    default:
      return null;
  }
}

export function getHeaderLayout(collection) {
  const {
    heroLayout,
    heroStyles,
    descriptionFormatted: description
  } = collection.attributes;
  if (isEmpty(heroStyles))
    return description ? "title_description" : "title_only";
  return heroLayout;
}
