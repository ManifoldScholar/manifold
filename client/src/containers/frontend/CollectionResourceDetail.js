import React, { PureComponent, PropTypes } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router';
import { Resource, Utility } from 'components/frontend';

import { entityStoreActions } from 'actions';
import { entityUtils } from 'utils';
import { collectionsAPI } from 'api';

const { select } = entityUtils;
const { request, flush, requests } = entityStoreActions;

class CollectionResourceDetailContainer extends PureComponent {
  static fetchData(getState, dispatch, location, params) {
    const page = params.page ? params.page : 1;
    const collectionFetch = collectionsAPI.show(params.collectionId);
    const collectionResourceFetch = collectionsAPI.collectionResource(
      params.collectionId,
      params.collectionResourceId
    );
    const collectionAction = request(collectionFetch, requests.feCollection);
    const collectionResourceAction = request(
      collectionResourceFetch,
      requests.feCollectionResources
    );
    const { promise: one } = dispatch(collectionAction);
    const { promise: two } = dispatch(collectionResourceAction);
    return Promise.all([one, two]);
  }

  static mapStateToProps(state) {
    const props = {
      collection: select(requests.feCollection, state.entityStore),
      collectionResource: select(requests.feCollectionResources, state.entityStore)
    };
    return props;
  }

  static propTypes = {
    project: PropTypes.object,
    resource: PropTypes.object
  };

  collectionUrl() {
    const cid = this.props.collection.id;
    const pid = this.props.collection.relationships.project.id;
    return `/browse/project/${pid}/collection/${cid}`;
  }

  render() {
    return (
      <div>
        {this.props.collection ?
          <Utility.BackLinkPrimary
            backText="Back to Collection"
            link={this.collectionUrl()}
            title={this.props.collection.attributes.title}
          /> : null
        }
        {this.props.collectionResource ?
          <Resource.Detail
            projectId={this.props.params.id}
            resource={this.props.collectionResource.relationships.resource}
          /> : null
        }
        {this.props.collection ?
          <section className="bg-neutral05">
            <Utility.BackLinkSecondary
              link={this.collectionUrl()}
              backText="Back to Collection"
            />
          </section> : null
        }

      </div>
    );
  }
}

const CollectionResourceDetail = connect(
  CollectionResourceDetailContainer.mapStateToProps
)(CollectionResourceDetailContainer);

export default CollectionResourceDetail;
