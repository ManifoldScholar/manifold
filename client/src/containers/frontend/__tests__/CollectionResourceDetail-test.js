import React from "react";
import renderer from "react-test-renderer";
import { CollectionResourceDetailContainer } from "../CollectionResourceDetail";
import { Provider } from "react-redux";
import build from "test/fixtures/build";
import { wrapWithRouter } from "test/helpers/routing";

describe("Frontend CollectionResourceDetail Container", () => {
  const store = build.store();

  const project = build.entity.project("1");
  const collection = build.entity.collection("2");
  const resource = build.entity.resource("3");
  const collectionResource = build.entity.collectionResource("4");
  collection.relationships.resources.push(resource);
  collectionResource.relationships.resource = resource;
  resource.relationships.collectionResources.push(collectionResource);
  project.relationships.resources.push(resource);

  const component = renderer.create(
    wrapWithRouter(
      <Provider store={store}>
        <CollectionResourceDetailContainer
          collection={collection}
          collectionResource={collectionResource}
          resource={resource}
          match={{
            params: {}
          }}
        />
      </Provider>
    )
  );

  it("renders correctly", () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("doesn't render to null", () => {
    let tree = component.toJSON();
    expect(tree).not.toBe(null);
  });
});
