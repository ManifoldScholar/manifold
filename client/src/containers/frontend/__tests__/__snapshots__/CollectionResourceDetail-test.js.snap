// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frontend CollectionResourceDetail Container renders correctly 1`] = `
<div>
  <a
    className="back-link-primary full"
    href="/project/1/collection/2"
    onClick={[Function]}
  >
    <div
      className="container flush"
    >
      <i
        className="manicon manicon-arrow-left"
      />
      Back to Collection
      <span>
        Rowan
      </span>
    </div>
  </a>
  <div>
    <section>
      <section
        className="resource-detail"
      >
        <div
          className="container flush-top flush-bottom"
        >
          <div
            className="resource-title"
          >
            <div>
              <h1
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Image",
                  }
                }
              />
              <span
                className="resource-date"
              >
                Resource added 
                <span>
                  April, 2017
                </span>
              </span>
            </div>
          </div>
        </div>
        <section
          className="resource-hero-container"
        >
          <div
            className="resource-hero"
          >
            <div
              className="resource-slide-figure"
            >
              <figure>
                <div
                  className="resource-link-wrapper"
                >
                  <span />
                  <div
                    className="resource-preview-wrapper"
                    onClick={[Function]}
                  >
                    <div
                      className="zoom-indicator"
                    >
                      Zoom
                      <i
                        className="manicon manicon-magnify-plus"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="figure-image"
                  style={
                    Object {
                      "backgroundImage": "url(null)",
                    }
                  }
                />
              </figure>
            </div>
          </div>
        </section>
        <div
          className="container flush-top"
        >
          <aside>
            <a
              className="button-primary"
              href="original-image.mock"
              target="_blank"
            >
              Download 
              <i
                className="manicon manicon-arrow-down"
              />
            </a>
            <nav
              className="share-nav-primary"
            >
              <span>
                Share
              </span>
              <ul>
                <li>
                  <button
                    onClick={[Function]}
                    target="_blank"
                  >
                    <i
                      className="manicon manicon-twitter"
                    />
                  </button>
                </li>
              </ul>
            </nav>
            <section
              className="resource-meta"
            >
              <figure
                className="resource-type"
              >
                <i
                  className="manicon manicon-resource-image"
                />
              </figure>
              <ul
                className="meta-list-secondary"
              >
                <li>
                  <span
                    className="meta-label"
                  >
                    Type
                  </span>
                  <span
                    className="meta-value"
                  >
                    Image
                  </span>
                </li>
                <li>
                  <span
                    className="meta-label"
                  >
                    Created On
                  </span>
                  <span
                    className="meta-value"
                  >
                    <span>
                      April 24, 2017
                    </span>
                  </span>
                </li>
              </ul>
              <nav
                className="resource-tag-list"
                onClick={[Function]}
                onMouseOver={[Function]}
              >
                <ul>
                  <li>
                    <a
                      className={null}
                      href="/project/1/resources?tag=dog"
                      onClick={[Function]}
                    >
                      dog
                    </a>
                  </li>
                  <li>
                    <a
                      className={null}
                      href="/project/1/resources?tag=puppy"
                      onClick={[Function]}
                    >
                      puppy
                    </a>
                  </li>
                  <li>
                    <a
                      className={null}
                      href="/project/1/resources?tag=goat"
                      onClick={[Function]}
                    >
                      GOAT
                    </a>
                  </li>
                </ul>
              </nav>
            </section>
          </aside>
          <div
            className="resource-meta-mobile"
          >
            <section
              className="resource-meta"
            >
              <figure
                className="resource-type"
              >
                <i
                  className="manicon manicon-resource-image"
                />
              </figure>
              <ul
                className="meta-list-secondary"
              >
                <li>
                  <span
                    className="meta-label"
                  >
                    Type
                  </span>
                  <span
                    className="meta-value"
                  >
                    Image
                  </span>
                </li>
                <li>
                  <span
                    className="meta-label"
                  >
                    Created On
                  </span>
                  <span
                    className="meta-value"
                  >
                    <span>
                      April 24, 2017
                    </span>
                  </span>
                </li>
              </ul>
              <nav
                className="resource-tag-list"
                onClick={[Function]}
                onMouseOver={[Function]}
              >
                <ul>
                  <li>
                    <a
                      className={null}
                      href="/project/1/resources?tag=dog"
                      onClick={[Function]}
                    >
                      dog
                    </a>
                  </li>
                  <li>
                    <a
                      className={null}
                      href="/project/1/resources?tag=puppy"
                      onClick={[Function]}
                    >
                      puppy
                    </a>
                  </li>
                  <li>
                    <a
                      className={null}
                      href="/project/1/resources?tag=goat"
                      onClick={[Function]}
                    >
                      GOAT
                    </a>
                  </li>
                </ul>
              </nav>
            </section>
          </div>
          <div
            className="resource-content left"
          >
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "World's Greatest Dog",
                }
              }
            />
            <h3
              className="attribute-header"
            >
              Full Description
            </h3>
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Black and white freckles",
                }
              }
            />
            <div
              className="resource-comments"
            >
              <div
                className="comment-editor"
              >
                <div
                  className="placeholder"
                >
                  <button
                    onClick={[Function]}
                  >
                    Login to post a comment
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </section>
  </div>
  <section
    className="bg-neutral05"
  >
    <div
      className="container flush"
    >
      <a
        className="back-link-secondary"
        href="/project/1/collection/2"
        onClick={[Function]}
      >
        <i
          className="manicon manicon-arrow-round-left"
        />
        <div>
          <span
            className="back-text"
          >
            Back to Collection
          </span>
          <span
            className="project-title"
          />
        </div>
      </a>
    </div>
  </section>
</div>
`;
